### PR TITLE
Medical - Disable blood spurts for fire, burn and drown damage types

### PR DESCRIPTION
**Pull request by veteran29**
**When merged this pull request will:**
- title
- Added `noBlood` config param to the damageTypes configs in ACE_Medical_Injuries
- Docs

### CODE REVIEW

**Comment by veteran29** [comment]
FIY @pterolatypus

File: addons/medical_blood/functions/fnc_handleWoundReceived.sqf [hunk 1]
```diff
@@ -18,8 +18,11 @@
  * Public: No
  */
 
-params ["_unit", "_allDamages", "_shooter"];
-(_allDamages select 0) params ["_damage", ""];
+params ["_unit", "_allDamages", "_shooter", "_damageType"];
+(_allDamages select 0) params ["_damage"];
+
+// Don't bleed if damage type does not cause bleeding
+if (GVAR(noBloodDamageTypes) getOrDefault [_damageType, false]) exitWith {};
```
Comment (mharis001) [inline addons/medical_blood/functions/fnc_handleWoundReceived.sqf:25]: I think use `in` here.

**Comment by pterolatypus** [comment]
ideally I would like to see this be based on wound type rather than damage type, but that would require a bit of restructuring

File: addons/medical_blood/functions/fnc_handleWoundReceived.sqf (hunk 1 above)
Comment (veteran29) [reply addons/medical_blood/functions/fnc_handleWoundReceived.sqf]: Yeah, I forgot it works for HashMaps

**Comment by veteran29** [comment]
> I'd be interested to know if there is actually any benefit to caching the config at prestart rather than just reading it fresh at preinit.

```sqf
// Execution Time: 0.0015 ms  |  Cycles: 10000/10000  |  Total Time: 15 ms
(call (uiNamespace getVariable "ace_medical_blood_noBloodDamageTypes"))
```

```sqf
// Execution Time: 0.0234 ms  |  Cycles: 10000/10000  |  Total Time: 234 ms
"getNumber (_x >> 'noBlood') == 1" configClasses (configFile >> "ACE_Medical_Injuries" >> "damageTypes") apply {[configName _x, nil]}
```

Is it worth it? Not sure but it's how we've been doing it in other components so doing it like this for consistency. Also, the more damage types the slower the config crawl so not doing that on every mission load is in theory beneficial if there's a lot of custom damage types (unlikely).

**Comment by pterolatypus** [comment]
fair enough, that's more than I was expecting

**Review by mharis001 — APPROVED** [review]
Code LGTM.

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**